packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java

```diff
@@ -147,6 +153,87 @@ public void chooseMultiImageFromGallery_WhenPendingResultExists_FinishesWithAlre
             any(Intent.class), eq(ImagePickerDelegate.REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY));
   }
 
+  @Test
+  @Config(minSdk = 33)
+  public void
+      chooseImageFromGallery_WithPhotoPicker_WhenHasExternalStoragePermission_LaunchesChooseFromGalleryIntent() {
+    when(mockPermissionManager.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE))
+        .thenReturn(true);
+
+    ImagePickerDelegate delegate = createDelegate();
+    delegate.chooseImageFromGallery(mockMethodCall, mockResult);
+
+    verify(mockActivity)
+        .startActivityForResult(
+            any(Intent.class),
+            eq(ImagePickerDelegate.REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY_USING_PHOTO_PICKER));
```

Comment:
(Continuing from https://github.com/flutter/plugins/pull/7043#discussion_r1113577211 on the pre-migration version of this PR, and [from Discord](https://discord.com/channels/608014603317936148/846507907876257822/1078039169319845939).)

This is checking what "request code" we passed to `startActivityForResult`. But the request code doesn't have any effect on what UI the system presents to the user, or anything like that — it's a number we make up for our own internal tracking, so that our `onActivityResult` can know what to do with the result it gets.

The thing that controls whether the system shows the spiffy new photo picker, or does something else, is the "action" in the Intent. So that would be the thing to inspect if writing this style of unit test.

Reply:
Here's probably the clearest upstream docs on that — in particular the "Action" subsection:
https://developer.android.com/guide/components/intents-filters

Then in the reference docs there's
https://developer.android.com/reference/android/app/Activity#startActivityForResult(android.content.Intent,%20int)
https://developer.android.com/reference/android/app/Activity#onActivityResult(int,%20int,%20android.content.Intent)
and the voluminous intro of
https://developer.android.com/reference/android/content/Intent.html
.

Reply:
> Here's probably the clearest upstream docs on that — in particular the "Action" subsection: https://developer.android.com/guide/components/intents-filters

Is this something we want want to implement for this PR then? Looks like all the tests currently only check that the intent has the correct tag so I imagine we'd want to refactor those tests to also confirm the correct action.

Reply:
> I imagine we'd want to refactor those tests to also confirm the correct action.

I think that would be a good improvement, but my suggestion for this PR would be to leave the tests for other functionality as they are.